internal/provider/resource_project_cloud_account_link.go

```diff
@@ -0,0 +1,668 @@
+package provider
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/google/uuid"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"wiz.io/hashicorp/terraform-provider-wiz/internal"
+	"wiz.io/hashicorp/terraform-provider-wiz/internal/client"
+	"wiz.io/hashicorp/terraform-provider-wiz/internal/utils"
+	"wiz.io/hashicorp/terraform-provider-wiz/internal/wiz"
+)
+
+// CloudAccountSearchResponse represents the response from a cloud account search.
+// It includes a GraphSearch object that contains a list of Nodes, each with a list of Entities.
+type CloudAccountSearchResponse struct {
+	GraphSearch struct {
+		Nodes []struct {
+			Entities []struct {
+				ID string `json:"id"`
+			} `json:"entities"`
+		} `json:"nodes"`
+	} `json:"graphSearch"`
+}
+
+// SearchForCloudAccountVars represents the variables for a cloud account search.
+// It includes the external ID, project ID, and a quick search flag.
+type SearchForCloudAccountVars struct {
+	ExternalID string `json:"externalId"`
+	ProjectID  string `json:"projectId"`
+	Quick      bool   `json:"quick"`
+}
+
+// PartialProjectWithCloudAccountLinks represents a partial project with cloud account links.
+// It includes a Project object.
+type PartialProjectWithCloudAccountLinks struct {
+	Project PartialProject `json:"project"`
+}
+
+// PartialProject represents a partial project.
+// It includes a list of cloud account links.
+type PartialProject struct {
+	CloudAccountLinks []*wiz.ProjectCloudAccountLink
+}
+
+// UpdateProjectCloudAccountLinks represents the input for updating project cloud account links.
+// It includes the ID and a Patch object.
+type UpdateProjectCloudAccountLinks struct {
+	ID    string                        `json:"id"`
+	Patch PatchProjectCloudAccountLinks `json:"patch"`
+}
+
+// PatchProjectCloudAccountLinks represents the patch object for updating project cloud account links.
+// It includes a list of wiz.ProjectCloudAccountLinkInput.
+type PatchProjectCloudAccountLinks struct {
+	CloudAccountLinks []*wiz.ProjectCloudAccountLinkInput `json:"cloudAccountLinks"`
+}
+
+func resourceWizProjectCloudAccountLink() *schema.Resource {
+	return &schema.Resource{
+		Description: "Please either use this resource or the embedded set of Cloud Account Links in the wiz_project resource. " +
```

Comment:
Consider updating to: Associate a cloud subscription with a project. Use **either** this resource or the `cloud_account_link` block set for the wiz_project, never both.